projects/eval-pqmon/src/common/afe_config.c

```diff
@@ -0,0 +1,514 @@
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+
+#include "afe_config.h"
+
+/******************************************************************************/
+/********************** Macros and Constants Definitions **********************/
+/******************************************************************************/
+
+volatile uint8_t rmsOneReady = 0;
+struct no_os_spi_desc *hSPI;
+struct no_os_spi_msg spiMsg;
+extern struct no_os_gpio_init_param reset_gpio_ip;
+
+/******************************************************************************/
+/************************* Functions Definitions ******************************/
+/******************************************************************************/
+
+int32_t afe_init(void)
+{
+	int32_t status = 0;
+	uint32_t config;
+
+	// status = timer_init();
+	// if (status)
+	// 	return status;
+	status = spi_device_init();
+	if (status)
+		return status;
+	status = reset_afe();
+	if (status)
+		return status;
+	if (status == 0) {
+		config = 0;
+		status = afe_write_16bit_reg(REG_RUN, (uint16_t *)&config);
+		if (status != 0) {
+			status = SYS_STATUS_AFE_RUN_FAILED;
+		}
+	}
+	config = BITM_CONFIG0_RMS_SRC_SEL | BITM_CONFIG0_PERIOD_AVG_CFG_AV16;
+	status = afe_write_32bit_reg(REG_CONFIG0, (uint32_t *)&config);
+	if (status != 0) {
+		status = SYS_STATUS_AFE_CONFIG0_FAILED;
+	} else {
+		config = BITM_CONFIG1_BURST_EN;
+		config |= BITM_CONFIG1_CF3_CFG;
+		status = afe_write_16bit_reg(REG_CONFIG1, (uint16_t *)&config);
+		if (status != 0) {
+			status = SYS_STATUS_AFE_CONFIG1_FAILED;
+		}
+	}
+
+	if (status == 0) {
```

Comment:
i see you're using the alternate method for handling return statuses

there is nothing wrong with it but it just looks at odds with the rest of code in the no-OS repo